charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -62,6 +62,20 @@ spec:
             - --logtostderr=true
             {{- end }}
             - --localhost=127.0.0.1
+            {{- if .Values.prometheusOperator.prometheusDefaultBaseImage }}
+            {{- if .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
+            - --prometheus-default-base-image={{ .Values.prometheusOperator.prometheusDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.tag }}@sha256:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
+            {{- else }}
+            - --prometheus-default-base-image={{ .Values.prometheusOperator.prometheusDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.tag }}
+            {{- end }}
+            {{- end }}
+            {{- if .Values.prometheusOperator.alertmanagerDefaultBaseImage }}
```

Comment:
Why don't you use "if and"?

Reply:
I am not sure how you would use if and as there is a nested if in the first one to set the sha if specified

Reply:
for example:

```
{{- if and .Values.prometheusOperator.prometheusDefaultBaseImage .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
```

```
{{- if and .Values.prometheusOperator.alertmanagerDefaultBaseImage .Values.prometheusOperator.alertmanagerDefaultBaseImage.sha }}
```

Reply:
I guess I could use the following instead of using nesting but this is duplicating the condition needlessly in my opinion:
```
{{- if and .Values.prometheusOperator.prometheusDefaultBaseImage
 .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
{{- else if .Values.prometheusOperator.prometheusDefaultBaseImage }}
{{- end }}
```

Reply:
@monotek I misunderstood how alertmanager-default-base-image was supposed to be configured as we should not set the tag and sha. I updated my PR to configure it as it should be